fvm/transactionVerifier.go

```diff
@@ -175,22 +186,18 @@ func (v *TransactionSignatureVerifier) verifyAccountSignature(
 		accountKey.HashAlgo,
 	)
 	if err != nil {
-		if errors.Is(err, ErrInvalidHashAlgorithm) {
-			return nil, &InvalidHashAlgorithmError{
-				Address:  txSig.Address,
-				KeyIndex: txSig.KeyIndex,
-				HashAlgo: accountKey.HashAlgo,
-			}
+		if sType == envelopeSignature {
+			return nil, errors.NewInvalidEnvelopeSignatureError(txSig.Address, txSig.KeyIndex, err)
 		}
-
-		return nil, err
+		return nil, errors.NewInvalidPayloadSignatureError(txSig.Address, txSig.KeyIndex, err)
```

Comment:
For the same reason as before, could we separate out the proposer signature being invalid?

Reply:
def we could and I think we should I just tried to not change the logic of this part in this PR. for the next PR I'm going to refactor txVerifier and add tests.

Reply:
actually we do have `InvalidProposalSignatureError`